### ISSUE TITLE
Adding an "expire on" feature

### ISSUE THREAD

**Issue by DiomFR** [jflesch/paperwork#610]
As Paperwork is mostly used to manage all papers we have, it might be a good idea to add an "expire on" feature (ie. keep a invoice until the warranty expires) to manually set when we add document.

And, of course, a way to show all the expired document to delete them manually.

**Comment by jflesch** [comment]
I'm trying to understand the use case here: Why would you want to delete expired documents ? oO
Do you have a concrete example ?

**Comment by ftorto** [comment]
Avoiding keeping too much old bills (warranty expired) can be a use case

**Comment by DiomFR** [comment]
Hello,

The first example I can think of is a paper you have to keep for a defined time. 

Like a invoice for an object with 2 years warranty, you don't want to keep it for ever. 
Or you're insurance card (carte mutuelle in France), you have a new one each year. 

Thanks for your work, I discovered it on linuxfr. 


Just a thing, to install an update, what's the process ?

Le 25 février 2017 12:19:10 GMT+01:00, Jerome Flesch <notifications@github.com> a écrit :
>I'm trying to understand the use case here: Why would you want to
>delete expired documents ? oO
>Do you have a concrete example ?
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/jflesch/paperwork/issues/610#issuecomment-282477654


--
Damien, sent from my mobile phone, sorry for my brevity.

**Comment by tiramiseb** [comment]
Why not keep it ? The longer the better.

Regarding invoices, you must keep them as long as possible, even after the 2-years delay : you may need the to prove you owned things after a house fire, for insurance...

I'm sure I can find cases like that for every document type you'll find :)

**Comment by jstenback** [comment]
I could see a use case for an "archive on" feature rather that would simply add a label that automatically hides documents from the default view. And a mass-delete feature could in an of itself be useful, but I certainly wouldn't want anything to ever disappear automatically to a point where documents wouldn't be easily recoverable.

**Comment by jflesch** [comment]
> you don't want to keep it for ever.

Why not ?

> automatically hides documents from the default view

Maybe am I being a little obtuse here but, I still don't see in which way it would be interesting to automatically hide a document after some times.

I mean:
* either you don't need the document ever again, in which case you can just keep it in your work directory forever and forget about it (Afaik, these is no reason for this document to bother you)
* or you do, in which case these features brings nothing

In short, I'm still waiting for a concrete use case.

**Comment by tiramiseb** [comment]
> I could see a use case for an "archive on" feature

errr... in my opinion, everything in paperwork is already "archived"... whatever is in paperwork is a part of the pile of archived documents I will maybe need to access once, sometimes later.

**Comment by lcecic** [comment]
Hello

My "expire on" use case : 
I use a owncloud to share the paperwork data on several computer. If I suppose to put document without deleting when they are useless, my owncloud share will necessarily get full, for the same reason I use to garbage things that become unnecessarily in my house (can we have non stop growth without get the resources full ?).
In france we have a document explaining how long we should get papers (sorry it's in french : https://www.service-public.fr/particuliers/vosdroits/F19134). So I can fill the "expire on" date for all the official paper. And, after a certain delay, get all my expired documents and delete then.

**Comment by jflesch** [comment]
>  (can we have non stop growth without get the resources full ?).

For reference, I have > 4 years worth of documents (1500 documents), and it takes only 2GB. I'm single (kinda), so with a family and a small business, I'm assuming you can multiply that by 5  => 10GB. Nowadays, 10GB is nothing.

The only case where this argument could make sense to me is if you don't host yourself but use a service like DropBox, and I would strongly advise against it.

**Comment by mjourdan** [comment]
With #119, it would be allowed to search "all the bills since last month" and save the request. Supporting something like "all the bills older than two years" would perfectly fulfill this ticket, I think.

On a code point of view, that would be neglictible work. The challenge is keeping a simple and clean UX, and see if the complexity added is worth the benefits.

I'll do mockups :)

**Comment by mjourdan** [comment]
Mockups available in #119. Feel free to take a look and comment!
